sopel/bot.py

```diff
@@ -489,6 +485,25 @@ def register(self, callables, jobs, shutdowns, urls):
                     callable_name,
                     interval)
 
+    def unregister_jobs(self, jobs):
+        for job in jobs:
+            callable_name = getattr(job, "__name__", 'UNKNOWN')
+            self.scheduler.remove_callable_job(job)
+            LOGGER.debug('Job callable removed: %s', callable_name)
+
+    def register_shutdowns(self, shutdowns):
+        # Append plugin's shutdown function to the bot's list of functions to
+        # call on shutdown
+        self.shutdown_methods = self.shutdown_methods + list(shutdowns)
```

Comment:
My Clojure/FP brain was switch on I guess.

Reply:
Also I just realized that we don't need this `shutdown_methods` list anymore, since we have a proper list of registered & loaded plugins, and we can retrieve all the shutdown methods when needed.

Hm... maybe I should deprecate that too.

Reply:
Probably, yeah. It always seemed kinda hacky, but especially so now that there's a proper plugin system.

Reply:
Let's do that in another PR.

Reply:
@dgw OK with that? :)